src/contracts/interfaces/IPaymentCoordinator.sol

```diff
@@ -0,0 +1,214 @@
+        IERC20 token;
+        uint256 amount;
+        uint32 startTimestamp;
+        uint32 duration;
+    }
+
+    struct DistributionRoot {
+        // merkle root of the distribution
+        bytes32 root;
+        // The timestamp until which payments have been calculated
+        uint32 paymentCalculationEndTimestamp;
+        // timestamp at which the root can be claimed against
+        uint32 activatedAt;
+    }
+
+    struct EarnerTreeMerkleLeaf {
+        address earner;
+        bytes32 earnerTokenRoot;
+    }
+
+    struct TokenTreeMerkleLeaf {
+        IERC20 token;
+        uint256 cumulativeEarnings;
+    }
+
+    struct PaymentMerkleClaim {
+        // The index of the root in the list of roots
+        uint32 rootIndex;
+        // proof of the earner's account root in the Earner Merkle tree
+        uint32 earnerIndex;
+        bytes earnerTreeProof;
+        EarnerTreeMerkleLeaf earnerLeaf;
+        // The indices and proofs of the leafs in the claimaint's merkle tree for this root
+        uint32[] tokenIndices;
+        bytes[] tokenTreeProofs;
+        TokenTreeMerkleLeaf[] tokenLeaves;
+    }
+
+    /// EVENTS ///
+
+    /// @notice emitted when an AVS creates a valid RangePayment
+    event RangePaymentCreated(
+        address indexed avs,
+        uint256 indexed paymentNonce,
+        bytes32 indexed rangePaymentHash,
+        RangePayment rangePayment
+    );
+    /// @notice emitted when a valid RangePayment is created for all stakers by a valid submitter
+    event RangePaymentForAllCreated(
+        address indexed submitter,
+        uint256 indexed paymentNonce,
+        bytes32 indexed rangePaymentHash,
+        RangePayment rangePayment
+    );
+    /// @notice paymentUpdater is responsible for submiting DistributionRoots, only owner can set paymentUpdater
+    event PaymentUpdaterSet(address indexed oldPaymentUpdater, address indexed newPaymentUpdater);
+    event PayAllForRangeSubmitterSet(
+        address indexed payAllForRangeSubmitter,
+        bool indexed oldValue,
+        bool indexed newValue
+    );
+    event ActivationDelaySet(uint32 oldActivationDelay, uint32 newActivationDelay);
+    event CalculationIntervalSecondsSet(uint32 oldCalculationIntervalSeconds, uint32 newCalculationIntervalSeconds);
+    event GlobalCommissionBipsSet(uint16 oldGlobalCommissionBips, uint16 newGlobalCommissionBips);
+    event ClaimerForSet(address indexed earner, address indexed oldClaimer, address indexed claimer);
+    /// @notice rootIndex is the specific array index of the newly created root in the storage array 
+    event DistributionRootSubmitted(
+        uint32 indexed rootIndex,
+        bytes32 indexed root,
+        uint32 paymentCalculationEndTimestamp,
+        uint32 activatedAt
+    );
+    /// @notice root is one of the submitted distribution roots that was claimed against
+    event PaymentClaimed(bytes32 indexed root, TokenTreeMerkleLeaf leaf);
```

Comment:
Can index the token address

`event PaymentClaimed(bytes32 indexed root, address indexed token, uint256 amount)`

Reply:
Do we want to update? @ypatil12 @gpsanant

Reply:
Just realized, we need to add the earner in this event too!

I'm fine with indexed or not indexed, but definitely need earner so the UI knows how much has been claimed already

Reply:
Based on standup today:

We need to emit (earner, claimer, claimedAmount, token, root). Leaf doesn't really matter anymore

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/514/commits/f13e1c21c5c4a90f1550fb80c132428bbacd731c
Updated and tested